### DISCUSSION TITLE
Loading namespaces based on route params?

### DISCUSSION THREAD

**Discussion by brattonross** [sergiodxa/remix-i18next#60] [Ideas]
Is there a way to load a namespace based on route params in v4?

In v3 we could use `params` from the loader function context and pass that to `getTranslations`. However, in v4 we provide namespaces as strings in the `handle` export, so we don't have access to the route params.

I think I can see it being possible to use the remix context in `entry.server.tsx` to add additional namespaces based on route matches, although I haven't looked into it too far. If it is possible, it would be nice if this library added an abstraction for this. e.g. being able to resolve a namespace via a function that gives access to the params:

```js
export const handle = {
  i18n: ({ params }) => params.feature
};
```

Am I missing a way to do this in v4? Would you be willing to add something like this to the library if not? (I am happy to contribute a PR for this if so)

**Comment by sergiodxa** [comment]
This sounds like a good idea, but I'm not really sure how it would work client side, if you have an idea on how to implement it, a PR is welcoming.
